### PR TITLE
trealla: init at 1.7.65

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 115380` at 70e3483d run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/efc01ac8624f5c4c00e1ed3ac432ab61">trealla</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/interpreters/trealla/default.nix:24:3:
```
   |
24 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/interpreters/trealla/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    sha256 = "sha256-e50IL6MW8PHTCqKZUlPwreUtm0cYhXuULenxkbxdZ0I=";
+  };
+
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace '-I/usr/local/include' "" \
+      --replace '-I/usr/local/lib' ""
+  '';
+
+  makeFlags = [ "GIT_VERSION=\"v${version}\"" ] ++ lib.optional stdenv.isDarwin "NOLDLIBS=1";
```
Comment (zseri) [inline pkgs/development/interpreters/trealla/default.nix:20]: Are you sure that this "setting" of `GIT_VERSION` does what you expect from it?

File: pkgs/development/interpreters/trealla/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  buildInputs = [ readline openssl ];
+
+  installPhase = ''
+    install -Dm755 -t $out/bin tpl
```
Comment (zseri) [inline pkgs/development/interpreters/trealla/default.nix:25]:
```suggestion
    install -D -t $out/bin tpl
```

File: pkgs/development/interpreters/trealla/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/interpreters/trealla/default.nix:20]: The line in https://github.com/infradig/trealla/blob/v1.7.35/Makefile#L1 sould be ?= IIRC. Good catch.

**Comment by siraben** [comment]
Looks like upstream has tests too, I'll enable `doCheck` locally and see what happens.

**Comment by siraben** [comment]
Don't we love floating point errors? 😆 Will open an issue upstream.

```ShellSession
trealla> 2c2
trealla> < [[3.0000000000000036,2.0000000000000009],[5.0000000000000053,0.99999999999999267],[-5.773159728050814e-15,1.0000000000000027],[0.99999999999999623,1.0000000000000036]]
trealla> ---
trealla> > [[3.0000000000000027,2.0000000000000009],[5.0000000000000036,0.99999999999999389],[-4.4408920985006262e-15,1.0000000000000024],[0.99999999999999756,1.0000000000000031]]
trealla> Running tests/issues/test082.pl ...
trealla> Running tests/issues/test088.pl ...
trealla> Running tests/issues/test091.pl ...
trealla> Running tests/issues/test093.pl ...
trealla> Running tests/issues/test094.pl ...
trealla> Running tests/issues/test108.pl ...
trealla> Running tests/issues/test116.pl ...
trealla> Running tests/issues/test132.pl ...
trealla> ============
trealla> TEST SUMMARY
trealla> ============
trealla> Failed: 1
trealla> Succeeded: 125
trealla> installing
```

**Comment by siraben** [comment]
Ok, enabled the tests and fixed a failed substitution in postPatch.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 115380` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>trealla</li>
  </ul>
</details>

File: pkgs/development/interpreters/trealla/default.nix (hunk 2 above)
Comment (siraben) [reply pkgs/development/interpreters/trealla/default.nix:29]: I prefer the former.

File: pkgs/development/interpreters/trealla/default.nix (hunk 1 above)
Comment (zseri) [reply pkgs/development/interpreters/trealla/default.nix]: nah, I looked at the GNU make manual, and gmake allows overriding variables assigned using `:=` on the command line. I don't know if any other make may be used. https://www.gnu.org/software/make/manual/html_node/Overriding.html#Overriding

File: pkgs/development/interpreters/trealla/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    substituteInPlace Makefile \
+      --replace '-I/usr/local/include' "" \
+      --replace '-L/usr/local/lib' "" \
+      --replace 'GIT_VERSION :=' 'GIT_VERSION ?='
+  '';
+
+  makeFlags = [ "GIT_VERSION=\"v${version}\""
+                (lib.optionalString withThread "THREADS=1")
+                (lib.optionalString (!withSSL) "NOSSL=1")
+              ] ++ lib.optional stdenv.isDarwin "NOLDLIBS=1";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/trealla/default.nix:24]:
```suggestion
  makeFlags = [
    "GIT_VERSION=\"v${version}\""
    (lib.optionalString withThread "THREADS=1")
    (lib.optionalString (!withSSL) "NOSSL=1")
  ] ++ lib.optional stdenv.isDarwin "NOLDLIBS=1";
```

Comment (zseri) [reply pkgs/development/interpreters/trealla/default.nix:24]:
```suggestion
  makeFlags = [
    "GIT_VERSION=\"v${version}\""
    (lib.optionalString withThread "THREADS=1")
    (lib.optionalString (!withSSL) "NOSSL=1")
    (lib.optionalString stdenv.isDarwin "NOLDLIBS=1")
  ];
```

**Pull request merged**